client_src/js/utils/validators/api.validator.js

```diff
@@ -0,0 +1,23 @@
+import { ClientError } from "../../exceptions/client-error.handler";
+import { ClientErrors } from "../../exceptions/client.exceptions";
+
+function validateServerResponse(response) {
+  // Don't really like this, there may be a better way but currently the server re-directs everything and was hard to detect otherwise.
```

Comment:
This function is not required and I intuitively find this bug prone // smelly. The `/api` now always responds with JSON, so no need to check DOCTYPE or object... I'd go for Content-Type to check for `application/json` ... if we must, but I dont think we do.

Reply:
I left it in because I'd triggered it using the system/info endpoint as that isn't under the /api route. 

Personally I think it's needed because if you miss-type a url, nothing is triggered in the form of an error check. 

I am however going to switch it to a different method.